redis_session.c

```diff
@@ -581,6 +581,59 @@ redis_session_key(redis_pool_member *rpm, const char *key, int key_len, int *ses
     return session;
 }
 
+/* {{{ PS_CREATE_SID_FUNC
+ */
+PS_CREATE_SID_FUNC(redis)
+{
+    int retries = 3;
+#if (PHP_MAJOR_VERSION < 7)
+    char *sid;
+#else
+    zend_string *sid;
+#endif
+
+    redis_pool *pool = PS_GET_MOD_DATA();
+
+    while (retries-- > 0) {
+#if (PHP_MAJOR_VERSION < 7)
+        sid = php_session_create_id((void **) &pool, newlen TSRMLS_CC);
+        redis_pool_member *rpm = redis_pool_get_sock(pool, sid TSRMLS_CC);
+#else
+        sid = php_session_create_id((void **) &pool TSRMLS_CC);
+        redis_pool_member *rpm = redis_pool_get_sock(pool, ZSTR_VAL(sid) TSRMLS_CC);
+#endif
+        RedisSock *redis_sock = rpm?rpm->redis_sock:NULL;
+
+        if (!rpm || !redis_sock) {
+            php_error_docref(NULL TSRMLS_CC, E_NOTICE,
+                "Redis not available while creating session_id");
+
+            efree(sid);
```

Comment:
You should use `zend_string_release` for PHP7

Reply:
Fixed